commands/cms/convertFields.ts

```diff
@@ -1,66 +1,79 @@
-// @ts-nocheck
-const path = require('path');
-const fs = require('fs');
-const { createIgnoreFilter } = require('@hubspot/local-dev-lib/ignoreRules');
-const { isAllowedExtension, getCwd } = require('@hubspot/local-dev-lib/path');
-const { logger } = require('@hubspot/local-dev-lib/logger');
-const { walk } = require('@hubspot/local-dev-lib/fs');
-const { getThemeJSONPath } = require('@hubspot/local-dev-lib/cms/themes');
-const { i18n } = require('../../lib/lang');
-const {
+import { Argv, ArgumentsCamelCase } from 'yargs';
+import path from 'path';
+import fs from 'fs';
+import { createIgnoreFilter } from '@hubspot/local-dev-lib/ignoreRules';
+import { isAllowedExtension, getCwd } from '@hubspot/local-dev-lib/path';
+import { logger } from '@hubspot/local-dev-lib/logger';
+import { walk } from '@hubspot/local-dev-lib/fs';
+import { getThemeJSONPath } from '@hubspot/local-dev-lib/cms/themes';
+import { i18n } from '../../lib/lang';
+import {
   FieldsJs,
   isConvertableFieldJs,
-} = require('@hubspot/local-dev-lib/cms/handleFieldsJS');
+} from '@hubspot/local-dev-lib/cms/handleFieldsJS';
+import { trackConvertFieldsUsage } from '../../lib/usageTracking';
+import { logError } from '../../lib/errorHandlers/index';
+import { EXIT_CODES } from '../../lib/enums/exitCodes';
+import { CommonArgs, YargsCommandModule } from '../../types/Yargs';
+import { makeYargsBuilder } from '../../lib/yargsUtils';
 
-const { trackConvertFieldsUsage } = require('../../lib/usageTracking');
-const { logError } = require('../../lib/errorHandlers/index');
-const { EXIT_CODES } = require('../../lib/enums/exitCodes');
+const command = 'convert-fields';
+const describe = i18n(`commands.convertFields.describe`);
 
-exports.command = 'convert-fields';
-exports.describe = i18n(`commands.convertFields.describe`);
+type ConvertFieldsArgs = CommonArgs & {
+  src: string;
+  fieldOptions: string[];
+};
 
-const invalidPath = src => {
+function invalidPath(src: string): void {
   logger.error(
     i18n(`commands.convertFields.errors.invalidPath`, {
       path: src,
     })
   );
   process.exit(EXIT_CODES.ERROR);
-};
+}
 
-exports.handler = async options => {
-  let stats;
-  let projectRoot;
-  let src;
+async function handler(
+  args: ArgumentsCamelCase<ConvertFieldsArgs>
+): Promise<void> {
+  let stats: fs.Stats | undefined;
```

Comment:
nit but you can use `?:` instead of `| undefined`

Reply:
I'm not sure what's going on with my TS config, but for some reason my IDE is telling me that it's not a valid syntax, even though it is. Tried restarting my ts server but no luck  🤷 

![image](https://github.com/user-attachments/assets/19a0d5db-3997-4a96-ac4e-55ef556892d6)

Reply:
That's not valid syntax for assignments.  Only types, function args, and types/interfaces.  

[Example](https://www.typescriptlang.org/play/?#code/PTAEDUEMBsEsBMBQAzArgOwMYBdYHt1Rk88AKSAfgC5QBnbAJ1nQHMBKUAb0UQF8eQEGAkTYAngAcApkRKgAvKC6IlSyjXpNWfAWChwkmaJFq1QAMTndVoCUwBukbDPV1GzFgG4diQQEl0RwNEaClsUFdND28gA)